t/t1032-mf-priority-update-bank.t

```diff
@@ -0,0 +1,127 @@
+	flux account add-user \
+		--username=user1 \
+		--userid=5001 \
+		--bank=B \
+		--max-active-jobs=3 \
+		--max-running-jobs=2
+'
+
+test_expect_success 'send flux-accounting DB information to the plugin' '
+	flux account-priority-update -p $(pwd)/FluxAccountingTest.db
+'
+
+test_expect_success 'submit one job under default bank, two under non-default bank' '
+	job1=$(flux python ${SUBMIT_AS} 5001 --urgency=0 sleep 30) &&
+	job2=$(flux python ${SUBMIT_AS} 5001 --setattr=bank=B --urgency=0 sleep 30) &&
+	job3=$(flux python ${SUBMIT_AS} 5001 --setattr=bank=B --urgency=0 sleep 30)
+'
+
+test_expect_success 'update of bank of pending job works' '
+	flux update $job1 bank=B &&
+	flux job wait-event -t 30 $job1 priority &&
+	flux job eventlog $job1 > eventlog.out &&
+	grep "attributes.system.bank=\"B\"" eventlog.out
+'
+
+test_expect_success 'trying to update to a bank user does not have access to fails in job.validate' '
+	test_must_fail flux update $job1 bank=C > invalid_bank.out 2>&1 &&
+	test_debug "cat invalid_bank.out" &&
+	grep "cannot find user/bank or user/default bank entry for uid:" invalid_bank.out
+'
+
+test_expect_success 'trying to update to a bank that does not exist fails in job.validate' '
+	test_must_fail flux update $job1 bank=foo > nonexistent_bank.out 2>&1 &&
+	test_debug "cat nonexistent_bank.out" &&
+	grep "cannot find user/bank or user/default bank entry for uid:" nonexistent_bank.out
+'
+
+test_expect_success 'update a job to another bank that is at max-active-jobs limit' '
+	job4=$(flux python ${SUBMIT_AS} 5001 --urgency=0 sleep 30) &&
+	test_must_fail flux update $job4 bank=B > max_active_jobs.out 2>&1 &&
+	test_debug "cat max_active_jobs.out" &&
+	grep "new bank is already at max-active-jobs limit" max_active_jobs.out &&
+	flux job cancel $job4
+'
+
+test_expect_success 'cancel one of the jobs so bank is not at max-active-jobs limit' '
+	flux job cancel $job3
+'
+
+test_expect_success 'update urgency of held jobs so they transition to RUN' '
+	flux job urgency $job1 default &&
+	flux job urgency $job2 default &&
+	flux job wait-event -t 10 $job1 alloc &&
+	flux job wait-event -t 10 $job2 alloc
+'
+
+test_expect_success 'update a job to another bank that is at max-run-jobs limit' '
+	job5=$(flux python ${SUBMIT_AS} 5001 --urgency=0 sleep 30) &&
+	test_must_fail flux update $job5 bank=B > max_run_jobs.out 2>&1 &&
+	test_debug "cat max_run_jobs.out" &&
+	grep "new bank is already at max-run-jobs limit" max_run_jobs.out &&
+	flux job cancel $job5
+'
+
+test_expect_success 'cancel jobs' '
+	flux job cancel $job1 &&
+	flux job cancel $job2
+'
+
+test_expect_success 'submit job under non-default bank' '
+	job6=$(flux python ${SUBMIT_AS} 5001 --setattr=bank=B --urgency=0 sleep 30)
+'
+
+test_expect_success 'updating job to default bank works' '
+	flux update $job6 bank=A &&
+	flux job wait-event -t 30 $job6 priority &&
+	flux job eventlog $job6 > eventlog.out &&
+	grep "attributes.system.bank=\"A\"" eventlog.out
+'
```

Comment:
Is there a way to also check that the mf_priority plugin also sees the updated bank?

Reply:
Yes there is, thanks for this suggestion! I added another check in this test file that queries the plugin's map and checks that the intended update is recognized:

```
test_expect_success 'check that plugin also sees the job update' '
	flux jobtap query mf_priority.so > query.json &&
	jq -e ".mf_priority_map[0].banks[0].bank_name == \"A\"" <query.json &&
	jq -e ".mf_priority_map[0].banks[0].cur_active_jobs == 1" <query.json &&
	jq -e ".mf_priority_map[0].banks[1].bank_name == \"B\"" <query.json &&
	jq -e ".mf_priority_map[0].banks[1].cur_active_jobs == 0" <query.json
'
```